moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java

```diff
@@ -406,6 +406,60 @@
     assertThat(moshi.adapter(c)).isSameAs(moshi.adapter(a));
   }
 
+  @Test public void jsonAdapterTakingJsonAdapterParameter() throws Exception {
+    Moshi moshi = new Moshi.Builder()
+        .add(new PointWriterAndReaderJsonAdapter())
+        .add(new JsonAdapterTakingJsonAdapterParameter())
+        .build();
+    JsonAdapter<Line> lineAdapter = moshi.adapter(Line.class);
+    Line line = new Line(new Point(5, 8), new Point(3, 2));
+    assertThat(lineAdapter.toJson(line)).isEqualTo("\"[5,8]..[3,2]\"");
+    assertThat(lineAdapter.fromJson("\"[5,8]..[3,2]\"")).isEqualTo(line);
+  }
+
+  static class JsonAdapterTakingJsonAdapterParameter {
+    @ToJson String lineToJson(Line line, JsonAdapter<Point> pointAdapter) throws IOException {
```

Comment:
There’s two ways to do `@ToJson`: 
1. Returning an intermediate value that’s itself converted (like here).
2. Getting a JsonWriter passed in and using that.

I’m actually thinking the first isn’t a valid uses case. The problem is that you’re converting a thing to JSON but returning a thing that isn’t, yielding double encoding. In the test case above that’s a JSON array inside of a string. There’s plausible ways to make this work but I think it’s going to be rare.

This problem is symmetric for `@FromJson`.

Any opinions on supporting `JsonAdapter` parameters _only_ for methods that take a `JsonReader` or `JsonWriter` ?